## T Test

  * In statistics, a t-test is a hypothesis test used to determine if there is a significant difference between the means of   two groups

In [1]:
import pandas as pd

In [4]:
df = pd.read_csv("carInsurance_data.csv")
df.head()

Id  Age         Job  Marital Education Default  Balance HHInsurance  \
0   1   32  management   single  tertiary      No     1218         Yes   
1   3   29  management   single  tertiary      No      637         Yes   
2   4   25     student   single   primary      No      373         Yes   
3   5   30  management  married  tertiary      No     2694          No   
4   6   32  technician   single  tertiary      No     1625          No   

  CarLoan Communication  LastContactDay LastContactMonth  NoOfContacts  \
0      No     telephone              28              jan             2   
1      No      cellular               3              jun             1   
2      No      cellular              11              may             2   
3      No      cellular               3              jun             1   
4      No      cellular              22              may             1   

   DaysPassed  PrevAttempts CarInsurance  
0          -1             0           No  
1         119             1          Yes  
2          -1             0          Yes  
3          -1             0           No  
4         109             1          Yes

## one-sample t-test

 * A one-sample t-test is a statistical test used to determine whether the mean of a single sample is significantly different from a known or hypothesized population mean. 

 - H0: μ = μ0  (Sample Mean = Population Mean)
 - H1: μ ≠ μ0   (Sample Mean != Population Mean)

#### Population Mean

In [25]:
df['Balance'].mean()  # Population 

1642.2185072610605

#### sample

In [32]:
a = df['Balance'].iloc[20:30]
a # this is sample 

20     900
21    3859
22       0
23    6981
24     842
25    1057
26     300
27       0
28     138
29     216
Name: Balance, dtype: int64

In [31]:
import pingouin as pg1
pg.ttest(df['Balance'].iloc[20:30],df['Balance'].mean(),alternative='two-sided',confidence=0.95)

T  dof alternative     p-val               CI95%  cohen-d  \
T-test -0.297856    9   two-sided  0.772574  [-187.77, 3046.37]  0.09419   

         BF10     power  
T-test  0.321  0.058238

## Two Sample T-Test


 - A two-sample t-test is a statistical test used to compare the means of two independent groups to determine if they are significantly different from each other.

 - H0 : μ1 = μ2   ( Sample1 Mean = Sample2 Mean)
 - H0 : μ1 != μ2   ( Sample1 Mean != Sample2 Mean)

In [33]:
# T-test
import pingouin as pg
pg.ttest(df['Age'].iloc[0:10],df['Age'].iloc[10:20],alternative='two-sided',confidence=0.95)

T  dof alternative     p-val            CI95%   cohen-d    BF10  \
T-test -3.67214   18   two-sided  0.001744  [-30.34, -8.26]  1.642231  19.372   

           power  
T-test  0.934497

 - Pvalue 0.00174 < 0.05 Accept H0

 - Accept Ho : there is relationship between two sample

## Paired Sample T test

 - This test is used when you have two sets of scores that are related in some way (e.g., before-and-after measurements on the same group). The paired samples t-test determines if the means of the paired observations are significantly different.

 - Null Hypothesis (H0): There is no significant difference between the means of the paired measurements.
 - Alternative Hypothesis (H1): There is a significant difference between the means of the paired measurements

## CHI- Square Test

 - This test is used when you want to assess whether the observed categorical data fits a theoretical or expected distribution.
 - The null hypothesis(H0): assumes that there is no significant difference between the observed and expected frequencies.
 - The alternative hypothesis(H1): suggests that there is a significant difference.

In [34]:
tab1 = pd.crosstab(df.Marital, df.Education)

In [35]:
from scipy.stats import chi2_contingency

In [36]:
chi2_contingency(tab1)

(79.0332069293724,
 2.791195155932752e-16,
 4,
 array([[ 47.37048294, 177.57919622, 131.05032084],
        [223.67916245, 838.51300236, 618.80783519],
        [122.95035461, 460.90780142, 340.14184397]]))

 - 2.79 < 0.05     so Accept the H0 : There is no relation between Marital & Education